homeassistant/components/generic/config_flow.py

```diff
@@ -391,3 +426,39 @@ async def async_step_init(
             ),
             errors=errors,
         )
+
+
+class CameraImagePreView(CameraImageView):
+    """Camera view to temporarily serve an image."""
+
+    name = "api:camera:imgepreview"
+    previews: dict[str, GenericCamera] = {}
+    initialized = False
+
+    def __init__(
+        self, hass: HomeAssistant, camera: GenericCamera, preview_id: str
+    ) -> None:
+        """Initialize a basic camera view."""
+        self.preview_id = preview_id
+        if not self.initialized:
```

Comment:
Just register the view once, don't do this kind of magic.

Reply:
> Just register the view once, don't do this kind of magic.

Please could you clarify?

I didn't want to leave the preview images left over at the end (hence saving the preview_id so that it could be removed during `mark_invalid` at the end).

Without this, I think old image previews would remain active until HA restart.  But if that is ok, I could remove/simplify it.

Reply:
Views are singletons, registered once.
The idea is that you register the view once, have the flow id in the path. When the request is received, lookup the flow from the flow id, then grab the camera object you need from the config flow object.